src/Mvc/Mvc.ViewFeatures/src/Infrastructure/HttpUriHelper.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Components.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    internal class HttpUriHelper : UriHelperBase
+    {
+        private HttpContext _context;
+
+        public HttpUriHelper()
+        {
+        }
+
+        public void InitializeState(HttpContext context)
+        {
+            _context = context;
+            InitializeState();
+        }
+
+        protected override void InitializeState()
+        {
+            SetAbsoluteBaseUri(GetContextBaseUri());
```

Comment:
Should this throw if uninitialized?

Reply:
I added it, although I don't know how I feel about it as we are the only ones that can initialize it. So it's not a user error but a framework error we could simply cover with tests.